vayesta/solver_rewrite/__init__.py

```diff
@@ -0,0 +1,96 @@
+from .ebcc import REBCC_Solver, UEBCC_Solver, EB_REBCC_Solver, EB_UEBCC_Solver
+from .fci import FCI_Solver, UFCI_Solver
+from .ebfci import EB_EBFCI_Solver, EB_UEBFCI_Solver
+from .hamiltonian import is_ham, is_uhf_ham, is_eb_ham, ClusterHamiltonian
+from .ccsd import RCCSD_Solver, UCCSD_Solver
+from .mp2 import RMP2_Solver, UMP2_Solver
+from .cisd import RCISD_Solver, UCISD_Solver
+from .tccsd import TRCCSD_Solver
+from .dump import DumpSolver
+
+def get_solver_class(ham, solver):
+    assert(is_ham(ham))
+    uhf = is_uhf_ham(ham)
+    eb = is_eb_ham(ham)
+    return _get_solver_class(uhf, eb, solver)
+
+def check_solver_config(is_uhf, is_eb, solver, log):
+    try:
+        solver = _get_solver_class(is_uhf, is_eb, solver)
+    except ValueError as e:
+        spinmessage = "unrestricted" if is_uhf else "restricted"
+        bosmessage = "coupled electron-boson" if is_eb else "purely electronic"
+
+        log.critical("Error; solver %s not available for %s %s systems", solver, spinmessage, bosmessage)
+        raise e
+
+
+def _get_solver_class(is_uhf, is_eb, solver):
```

Comment:
Maybe this would be cleaner with a dictionary: `solver_dict[(solver, is_uhf, is_eb)]`

Reply:
That could work, though giving good error messages for the different cases (unrecognised solver, not implemented for uhf, or not implemented for eb) might be more difficult for a dictionary. I might leave it for now?

Reply:
That would be neat, but the issue becomes providing the correct errors for different situations (unknown solver, not implemented for uhf or eb)